.github/workflows/lint.yaml

```diff
@@ -1,37 +1,38 @@
 name: Lint
+
 on:
   push:
-    branches:
-      - main
+    branches: [main]
   pull_request:
-    branches:
-      - main
+    branches: [main]
+
+permissions:
+  contents: read
 
-jobs:    
-  style:
-    name: Check style with ruff and black
+jobs:
+  lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - name: Check out repo
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
         with:
-          python-version: 3.9
-      - name: style
-        run: |
-          pip3 install ruff==0.0.265 black==22.10.0
-          black --check bofire tests
-          ruff check .
-
-  pyright:
-    name: Typechecking with pyright
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+          python-version: "3.9"
+
+      - name: Set up uv
+        uses: astral-sh/setup-uv@v2
+        with:
+          enable-cache: true
+
+      - name: Install dependencies
+        run: uv pip install pre-commit ".[all]" --system
+
+      - name: Run pre-commit
+        run: pre-commit run --all-files --show-diff-on-failure
+
+      - name: Pyright
```

Comment:
ideally we could move this inside the pre-commit action but there appear to be issues with getting pyright to work with pre-commit